sklearn/preprocessing/_data.py

```diff
@@ -748,6 +762,11 @@ def partial_fit(self, X, y=None):
                                 estimator=self, dtype=FLOAT_DTYPES,
                                 force_all_finite='allow-nan')
 
+        if sample_weight is not None:
+            sample_weight = _check_sample_weight(sample_weight, X,
+                                                 dtype=X.dtype)
+            sample_weight = np.asarray(sample_weight)
```

Comment:
I am surprised _check_sample_weight does not convert to array already

Reply:
Reading the source code of `_check_sample_weight` it seems that it always does. Are you sure this line is needed @maikia?